book/src/dev/rfcs/0008-release-planning.md

```diff
@@ -0,0 +1,190 @@
+Broadly speaking, possible processes can be divided into three categories:
+
+1.  Do not expect our users to update their software and do not provide them a
+    means to do so;
+
+2.  Use an existing update / deployment mechanism to distribute our software;
+
+3.  Write our own update / deployment mechanism to distribute our software.
+
+The first category is mentioned for completeness, but we need to provide users
+with a way to update their software.  Unfortunately, this means that standalone
+binaries without an update mechanism are not a workable option for us.  The
+third category is also unfavorable, because it creates a large amount of work
+for a task that is not really the focus of our product.  This suggests that we
+focus on solutions in the second category.
+
+One solution in the second category is to publish Docker images.  This has a
+number of attractive features.  First, we already produce Docker images for our
+own cloud deployments, so there is little-to-no marginal effort required to
+produce these for others as a distribution mechanism.  Second, providing Docker
+images will make it easier for us to provide a collection of related software
+in the future (e.g., providing an easy-to-deploy Prometheus / Grafana instance,
+or a sidecar Tor instance).  Third, Docker has a solid upgrade story, and we
+can instruct users to use the `:latest` version of the Docker image or steer
+them to auto-update mechanisms like Watchtower.
+
+While this solution works well for cloud deployments, Docker is not suitable
+everywhere.  What should we do outside of Docker?  One solution would be to try
+to create packages for each platform-specific package manager (Homebrew,
+something for Windows, various different flavors of Linux distribution), but
+this creates a large amount of additional work requiring platform-specific
+knowledge.  Worse, this work cannot be outsourced to others without giving up
+control over our software distribution -- if, for instance, a third party
+creates a Homebrew package, and we recommend people install Zebra using that
+package, we're reliant on that third party to continue packaging our software
+forever, or leave our users stranded.
+
+Instead, we can publish `zebrad` as a Rust crate and recommend `cargo install`.
+This approach has two major downsides.  First, installation takes longer,
+because Zebra is compiled locally.  Second, as long as we have a dependency on
+`zcashconsensus`, we'll have to instruct users to install some
+platform-specific equivalent of a `build-essential` package.  However, even for
+crates such as `zcashconsensus` that build native code, the `cargo`-managed
+build process is far, far more reliable than build processes for C or C++
+projects.  We would not be asking users to run autotools or `./configure`, just
+a one-step `cargo install`.  We also know that it's possible to reliably build
+Zebra on each platform with minimal additional steps, because we do so in our
+CI.
```

Comment:
we do build inside the docker and it was a hassle to find the right deps

```docker
RUN apt-get update && \
	apt-get install -y --no-install-recommends \
	make cmake g++ gcc llvm libclang-dev
```

Reply:
I'm glad we check dependencies already.

I think we should explicitly talk about testing our dependency set, and how our docker builds make sure we're aware of any extra dependencies.

Reply:
We might be able to use @sunshowers' https://github.com/facebookincubator/cargo-guppy to manage changes to our dependency set in a more principled way.

So far, our plan for limiting non-Rust dependencies has just been to be generally allergic to them and make high-level choices that avoid them (e.g., sled instead of rocksdb, etc), and that's worked out OK so far.  But it's obviously not ideal.